nip47/event_handler_test.go

```diff
@@ -253,121 +307,213 @@ func TestHandleResponse_NoPermission(t *testing.T) {
+	assert.NoError(t, err)
 
 	appPermission := &db.AppPermission{
 		AppId: app.ID,
 		App:   *app,
-		Scope: constants.PAY_INVOICE_SCOPE,
+		Scope: constants.GET_BALANCE_SCOPE,
 	}
 	err = svc.DB.Create(appPermission).Error
 	assert.NoError(t, err)
 
+	content := map[string]interface{}{
+		"method": models.GET_BALANCE_METHOD,
+	}
+
 	payloadBytes, err := json.Marshal(content)
 	assert.NoError(t, err)
 
-	msg, err := nip04.Encrypt(string(payloadBytes), ss)
+	msg, err := cipher.Encrypt(string(payloadBytes))
 	assert.NoError(t, err)
 
 	reqEvent := &nostr.Event{
 		Kind:      models.REQUEST_KIND,
-		PubKey:    reqPubkey,
-		CreatedAt: nostr.Timestamp(time.Now().Add(time.Duration(-6) * time.Hour).Unix()),
+		CreatedAt: nostr.Now(),
 		Tags:      nostr.Tags{},
 		Content:   msg,
 	}
-	err = reqEvent.Sign(reqPrivateKey)
+
+	if version != "0.0" {
+		reqEvent.Tags = append(reqEvent.Tags, []string{"v", version})
+	}
+
+	err = reqEvent.Sign(reqPrivateKey2)
 	assert.NoError(t, err)
 
+	// set a different pubkey (this will not pass validation)
+	reqEvent.PubKey = reqPubkey
+
 	relay := tests.NewMockRelay()
 
 	nip47svc.HandleEvent(context.TODO(), relay, reqEvent, svc.LNClient)
 
-	// it shouldn't return anything for an old request
-	assert.Nil(t, relay.PublishedEvent)
+	assert.Nil(t, relay.PublishedEvents)
+}
 
-	// change the request to now
-	reqEvent.CreatedAt = nostr.Now()
+func TestHandleResponse_NoApp(t *testing.T) {
+	defer tests.RemoveTestService()
+	svc, err := tests.CreateTestService()
+	require.NoError(t, err)
+	nip47svc := NewNip47Service(svc.DB, svc.Cfg, svc.Keys, svc.EventPublisher)
+
+	reqPrivateKey := nostr.GeneratePrivateKey()
+	reqPubkey, err := nostr.GetPublicKey(reqPrivateKey)
+	assert.NoError(t, err)
+
+	app, cipher, err := tests.CreateAppWithPrivateKey(svc, reqPrivateKey, "1.0")
+	assert.NoError(t, err)
+
+	// delete the app
+	err = svc.DB.Delete(app).Error
+	assert.NoError(t, err)
+
+	content := map[string]interface{}{
+		"method": models.GET_BALANCE_METHOD,
+	}
+
+	payloadBytes, err := json.Marshal(content)
+	assert.NoError(t, err)
+
+	msg, err := cipher.Encrypt(string(payloadBytes))
+	assert.NoError(t, err)
+
+	reqEvent := &nostr.Event{
+		Kind:      models.REQUEST_KIND,
+		PubKey:    reqPubkey,
+		CreatedAt: nostr.Now(),
+		Tags:      nostr.Tags{[]string{"v", "1.0"}},
+		Content:   msg,
+	}
 	err = reqEvent.Sign(reqPrivateKey)
 	assert.NoError(t, err)
 
+	relay := tests.NewMockRelay()
+
 	nip47svc.HandleEvent(context.TODO(), relay, reqEvent, svc.LNClient)
-	assert.NotNil(t, relay.PublishedEvent)
+
+	// it shouldn't return anything for an invalid app key
+	assert.Nil(t, relay.PublishedEvents)
 }
 
-func TestHandleResponse_IncorrectPubkey(t *testing.T) {
+func TestHandleResponse_IncorrectVersions(t *testing.T) {
 	defer tests.RemoveTestService()
 	svc, err := tests.CreateTestService()
 	require.NoError(t, err)
+
+	doTestHandleResponse_IncorrectVersion(t, svc, "0.0", "1.0")
+	doTestHandleResponse_IncorrectVersion(t, svc, "1.0", "0.0")
```

Comment:
would it be worth adding a third test where NIP-44 is used but no version tag is passed?

Reply:
Done!